hazelcast/src/main/java/com/hazelcast/cache/impl/CacheProxyUtil.java

```diff
@@ -240,7 +254,7 @@ protected static int getPartitionId(NodeEngine nodeEngine, Data key) {
      */
     public static <V> void validateConfiguredValueType(Class<V> valueType, V value)
             throws ClassCastException {
-        if (Object.class != valueType) {
+        if (value != null && valueType != null && Object.class != valueType) {
             //means type checks required
             if (!valueType.isAssignableFrom(value.getClass())) {
                 throw new ClassCastException("Value " + value + "is not assignable to " + valueType);
```

Comment:
I think, type of value should be logged to give more details since `int value  = 1` and `String value = "1"` are same output `1` but their types are different